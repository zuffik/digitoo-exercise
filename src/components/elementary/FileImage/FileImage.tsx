import * as React from 'react';

interface Props extends Omit<React.ImgHTMLAttributes<HTMLImageElement>, 'src'> {
    file: File;
}

export const FileImage: React.FC<Props> = ({file, ...props}: Props) => {
    const src = React.useMemo(() => URL.createObjectURL(file), [file])
    return (
        <img {...props} alt={props.alt} src={src}/>
    );
};
