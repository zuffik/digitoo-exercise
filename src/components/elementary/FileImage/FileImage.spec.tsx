import * as React from 'react';
import {render} from '@testing-library/react'
import {FileImage} from "./FileImage";

describe('<FileImage/>', () => {
    it('should render', () => {
        URL.createObjectURL = jest.fn(() => 'source');
        const spy = jest.spyOn(URL, 'createObjectURL');
        const file = new File([], 'image.png', {type: 'image/png'});
        const {asFragment, getByAltText} = render(<FileImage file={file} alt="id"/>);
        expect(getByAltText('id')).toHaveAttribute('src', 'source');
        expect(spy).toHaveBeenCalledWith(file);
        expect(asFragment()).toMatchSnapshot();
    });
});
