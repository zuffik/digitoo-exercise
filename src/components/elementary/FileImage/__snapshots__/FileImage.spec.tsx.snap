// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileImage/> should render 1`] = `
<DocumentFragment>
  <img
    alt="id"
    src="source"
  />
</DocumentFragment>
`;
